### ISSUE TITLE
Anchor tags should not be used as buttons

### ISSUE THREAD

**Issue by alvaromontoro** [jackdomleo7/Checka11y.css#104]
### Describe the new a11y feature or project enhancement

Anchor tags should not be used as buttons, it is bad for accessibility and usability. Links should redirect to a resource/page, if they don't they probably should be buttons.

### Describe the solution you'd like

Display a warning if a link is being used as a button. Examples of links being used as buttons:

```
<a href="#">Do something</a>

<a href="javascript:void(0)" role="button">Do another thing</a>

<a href="javascript:doSomething(0)">Do another thing</a>
```

Basically, what should trigger the warning would be:

- An anchor tag with `href="#"` (a link that doesn't go anywhere)
- An anchor tag with `href` starting with "javascript:" (a link that calls a function instead of redirecting)
- As a bonus, any anchor tag (or any element different than a button or input) that has `role="button"` should probably display the warning.

### Link(s)

- [The Anchor Button: Bad for Accessibility, Bad for Usability](https://itstiredinhere.blogspot.com/2014/08/the-anchor-button-bad-for-accessibility.html)
- [Proper Use of Buttons and Links](https://www.webaxe.org/proper-use-buttons-links/)
- [Anchor tag used as button](https://www.htmhell.dev/8-anchor-tag-used-as-button/)

**Comment by jackdomleo7** [comment]
I'm fully behind this idea. I think also from a DX point of view, seeing anchors can be quite confusing when they're acting as buttons. I also agree that this should be a warning.

**Comment by alvaromontoro** [comment]
It seems like a simple change, good for someone looking for a first contribution.

I'll pick it if no one does after a while.

**Comment by Matrix278** [comment]
Hi, can I try to work on this feature?

**Comment by jackdomleo7** [comment]
@Matrix278 absolutely!

**Comment by Matrix278** [comment]
Do you have anything about how the branch should be named?

**Comment by jackdomleo7** [comment]
@Matrix278 not really, any naming is fine - as long as it describes the feature somewhat 🙂
